### DISCUSSION TITLE
CIP Proposal - Finalize Consensus

### DISCUSSION THREAD

**Discussion by augustoteixeira** [cartesi/cips#29] [General]
We would like to propose a new system for finalizing consensus on Rollups. This would allow in the future for Validators to revert fake claims made by their nodes in case of an attack. This is further explained in the file linked below.

**Comment by ZzzzHui** [comment]
@augustoteixeira maybe you forgot to link the file :)

**Reply by augustoteixeira** [reply]
The discussion got much more elaborate than I had anticipated. So the file is quite outdated :(

But I feel like it is converging.
